tools/git.py

```diff
@@ -31,27 +33,44 @@ def __init__(
         :param commit: Git commit hash
         :param checkout_folder_name: Name of the output path repository folder
         :param clone_submodules: Flag to clone submodules if set to True
+        :param files_to_checkout: List of files to be checked out sparsely. If not set, the entire repo is checked out (default behaviour).
```

Comment:
Would the user need to use multiple files from different commits at some point? Maybe this parameter can be specified as a dictionary of commits to lists of files. Would that be confusing for the user?

Reply:
I think that is a bit too much for my use-case because all the commits are specified at repo-level. I will not add this now, but if everyone thinks it is useful, please let me know.